lib/node_modules/@stdlib/utils/async/parallel/test/test.main.js

```diff
@@ -0,0 +1,447 @@
+		t.throws( badValue( values[i] ), TypeError, 'throws a type error when provided ' + values[i] );
+	}
+	t.end();
+	function badValue( value ) {
+		return function badValue() {
+			parallel( [noop], value );
+		};
+	}
+});
+
+tape( 'the function throws an error if not provided a callback argument which is a function (options)', function test( t ) {
+	var values;
+	var i;
+
+	values = [
+		'5',
+		5,
+		NaN,
+		null,
+		true,
+		void 0,
+		[],
+		{}
+	];
+
+	for ( i = 0; i < values.length; i++ ) {
+		t.throws( badValue( values[i] ), TypeError, 'throws a type error when provided ' + values[i] );
+	}
+	t.end();
+	function badValue( value ) {
+		return function badValue() {
+			parallel( [noop], {}, value );
+		};
+	}
+});
+
+tape( 'the function throws an error if provided an `options` argument which is not an object', function test( t ) {
+	var values;
+	var i;
+
+	function next( value, clbk ) {
+		clbk( null, value );
+	}
+
+	function foo( resolve ) {
+		resolve();
+	}
+
+	values = [
+		'5',
+		5,
+		NaN,
+		true,
+		false,
+		null,
+		void 0,
+		[],
+		function noop() {}
+	];
+
+	for ( i = 0; i < values.length; i++ ) {
+		t.throws( badValue( values[i] ), TypeError, 'throws a type error when provided '+values[i] );
+	}
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			parallel( [foo], value, next );
+		};
+	}
+});
+
+tape( 'the function throws an error if provided an invalid option', function test( t ) {
+	var values;
+	var i;
+
+	function next( value, clbk ) {
+		clbk( null, value );
+	}
+
+	function foo( resolve ) {
+		resolve();
+	}
+
+	values = [
+		'5',
+		-5,
+		3.14,
+		0,
+		NaN,
+		true,
+		false,
+		null,
+		void 0,
+		[],
+		{},
+		function noop() {}
+	];
+
+	for ( i = 0; i < values.length; i++ ) {
+		t.throws( badValue( values[i] ), TypeError, 'throws a type error when provided '+values[i] );
+	}
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			var opts = {
+				'limit': value
+			};
+			parallel( [foo], opts, next );
+		};
+	}
+});
+tape( 'The function invokes the provided functions once and return the results in order (asynchronous)', function test( t ) {
+	var expected;
+
+	function one( resolve ) {
+		setTimeout( onTimeout, 100 );
+		function onTimeout() {
+			resolve( null, 1 );
+		}
+	}
+
+	function two( resolve ) {
+		setTimeout( onTimeout, 400 );
+		function onTimeout() {
+			resolve( null, 2 );
+		}
+	}
+
+	function three( resolve ) {
+		setTimeout( onTimeout, 200 );
+		function onTimeout() {
+			resolve( null, 3 );
+		}
+	}
+
+	parallel( [one, two, three], done );
+
+	expected = [
+		1,
+		2,
+		3
+	];
+
+	function done( error, out ) {
+		if ( error ) {
+			t.fail( error.message );
+		} else {
+			t.pass( 'did not return an error' );
+		}
+		t.deepEqual( out, expected, 'returns expected output' );
+		t.end();
+	}
+});
+
+tape( 'The function invokes the provided functions once and return the results in order (synchronous)', function test( t ) {
```

Comment:
```suggestion
tape( 'the function invokes the provided functions once and return the results in order (synchronous)', function test( t ) {
```

Always lowercase the first letter of test descriptions here and elsewhere.